sklearn/utils/_indexing.py

```diff
@@ -560,11 +576,21 @@ def resample(*arrays, replace=True, n_samples=None, random_state=None, stratify=
     # convert sparse matrices to CSR for row-based indexing
     arrays = [a.tocsr() if issparse(a) else a for a in arrays]
     resampled_arrays = [_safe_indexing(a, indices) for a in arrays]
+
+    if sample_weight is not None and not use_weights_in_resampling:
+        resampled_weights = _safe_indexing(sample_weight, indices)
+
     if len(resampled_arrays) == 1:
         # syntactic sugar for the unit argument case
-        return resampled_arrays[0]
+        if sample_weight is not None and not use_weights_in_resampling:
+            return [resampled_arrays[0], resampled_weights]
+        else:
+            return resampled_arrays[0]
     else:
-        return resampled_arrays
+        if sample_weight is not None and not use_weights_in_resampling:
+            return [resampled_arrays, resampled_weights]
+        else:
+            return resampled_arrays
```

Comment:
This function should now always return `resampled_arrays` and never attempt to compute `resampled_weights`.